### PR TITLE
bump psalm, fix errors, add usePhpDocPropertiesWithoutMagicCall

### PR DESCRIPTION
**Pull request by orklah**
Bumping psalm on last version to use usePhpDocPropertiesWithoutMagicCall.

This allows to drop some ignored errors and gain some type coverage.

Also fixed some errors

### CODE REVIEW

**Comment by orklah** [comment]
This PR doesn't fix all current Psalm issues, but after seeing https://github.com/Roave/BetterReflection/pull/543#issuecomment-587856388, I prefer creating it now to avoid doing this work twice

File: src/Reflection/ReflectionProperty.php [hunk 1]
```diff
@@ -317,7 +317,7 @@ public function __clone()
     /**
      * @param object|null $object
      *
-     * @return mixed
+     * @return Closure(object, string): mixed
```
Comment (Ocramius) [inline src/Reflection/ReflectionProperty.php]: This isn't returning a callable

Comment (orklah) [reply src/Reflection/ReflectionProperty.php]: Indeed. Must have overlooked __invoke call at the end.

**Comment by orklah** [comment]
This is now error free on Psalm 3.9.2

FYI, I added those errors in baseline:
```
ERROR: InvalidReturnType - src\Reflection\Adapter\ReflectionClass.php:228:21 - The declared return type 'list<ReflectionProperty>' for Roave\BetterReflection\Reflection\Adapter\ReflectionClass::getProperties is incorrect, got 'array
<string, Roave\BetterReflection\Reflection\Adapter\ReflectionProperty>'
    public function getProperties($filter = null)


ERROR: InvalidReturnType - src\Reflection\Adapter\ReflectionObject.php:202:21 - The declared return type 'list<ReflectionProperty>' for Roave\BetterReflection\Reflection\Adapter\ReflectionObject::getProperties is incorrect, got 'arr
ay<string, Roave\BetterReflection\Reflection\Adapter\ReflectionProperty>'
    public function getProperties($filter = null)


ERROR: PossiblyNullPropertyAssignmentValue - src\Reflection\ReflectionFunctionAbstract.php:512:35 -  with non-nullable declared type 'array<array-key, PhpParser\Node\Stmt>' cannot be assigned nullable type 'array<array-key, PhpParse
r\Node\Stmt>|null'
        $this->getNode()->stmts = $functionNode->getStmts();


ERROR: PossiblyNullPropertyAssignmentValue - src\Reflection\ReflectionFunctionAbstract.php:524:35 -  with non-nullable declared type 'array<array-key, PhpParser\Node\Stmt>' cannot be assigned nullable type 'array<array-key, PhpParse
r\Node\Stmt>|null'
        $this->getNode()->stmts = $this->loadStaticParser()->parse('<?php ' . $newBody);
```

First and second one are due to a more precise definition of Reflection library by Psalm. Error type are not compatible between native library and BR (cf #530)

Third and fourth are due to an assignment in Php-Parser from Php-Parser and one side of the assignment could contain null. This is probably due to an error in phpdoc on Php-Parser side.

**Comment by Ocramius** [comment]
Thanks @orklah!

**Pull request merged**